### PR TITLE
admin login

### PR DESCRIPTION
**Pull request by ce-walalang**
PR for #21

**Summary of changes**
* Installed and configured Devise gem with corresponding views
* Assigned `/admin` as Devise path
* Added `/admin/dashboard` page; authenticate all access to it

### CODE REVIEW

File: spec/support/factory_bot.rb [hunk 1]
```diff
@@ -0,0 +1,3 @@
+RSpec.configure do |config|
+  config.include FactoryBot::Syntax::Methods
```
Comment (egiurleo) [inline spec/support/factory_bot.rb:2]: Woohoo thanks for adding!

File: app/helpers/api/admins_helper.rb [hunk 2]
```diff
@@ -0,0 +1,2 @@
+module Api::AdminsHelper
```
Comment (egiurleo) [inline app/helpers/api/admins_helper.rb:1]: Could you explain what this module will be used for?

File: app/models/admin.rb [hunk 3]
```diff
@@ -0,0 +1,2 @@
+class Admin < ApplicationRecord
```
Comment (egiurleo) [inline app/models/admin.rb:1]: Could you explain what this model will be used for?

File: spec/views/api/admins/passwords.html.erb_spec.rb [hunk 4]
```diff
@@ -0,0 +1,5 @@
+require 'rails_helper'
+
+RSpec.describe "admins/passwords.html.erb", type: :view do
+  pending "add some examples to (or delete) #{__FILE__}"
```
Comment (egiurleo) [inline spec/views/api/admins/passwords.html.erb_spec.rb:4]: Are you anticipating examples to add to this spec or should we delete it?

**Review by egiurleo — COMMENTED** [review]
Woohoo!!! This is so exciting. It looks great, I just had a couple questions about files that don't look like they're doing anything but might be important to Devise. Once those questions are answered I'm happy to approve.

Another question I have: what do you think is the best way to ensure that only the users we want will be able to access the admin stuff? Should we allow users to have a "role" and only allow admins to view the dashboard? (This can be a separate ticket/PR, I just want to understand your vision!)

File: app/helpers/api/admins_helper.rb (hunk 2 above)
Comment (ty18881) [reply app/helpers/api/admins_helper.rb:1]: This is obsolete.  I was messing around before following a tutorial and we decided to go with User as the model.

File: app/models/admin.rb (hunk 3 above)
Comment (ty18881) [reply app/models/admin.rb:1]: This is obsolete.  I was messing around before following a tutorial and we decided to go with User as the model.
I'm not exactly sure how to get rid of this though.  Any suggestions?

File: spec/views/api/admins/passwords.html.erb_spec.rb (hunk 4 above)
Comment (ty18881) [reply spec/views/api/admins/passwords.html.erb_spec.rb:4]: We can delete this.

File: app/helpers/api/admins_helper.rb (hunk 2 above)
Comment (ty18881) [reply app/helpers/api/admins_helper.rb:1]: Is it as simple as git rm?

File: app/models/admin.rb (hunk 3 above)
Comment (ty18881) [reply app/models/admin.rb:1]: Is it as simple as git rm on the file?

Comment (egiurleo) [reply app/models/admin.rb:1]: Yup exactly! You can call that command for any files you want to remove from this PR.

**Comment by ce-walalang** [comment]
> Woohoo!!! This is so exciting. It looks great, I just had a couple questions about files that don't look like they're doing anything but might be important to Devise. Once those questions are answered I'm happy to approve.
> 
> Another question I have: what do you think is the best way to ensure that only the users we want will be able to access the admin stuff? Should we allow users to have a "role" and only allow admins to view the dashboard? (This can be a separate ticket/PR, I just want to understand your vision!)

thanks, Emily! adding roles sounds great. maybe use the gem Pundit?

**Comment by egiurleo** [comment]
Closing in favor of #45.

**Pull request closed without merging**